array-reduce/main.js

```diff
@@ -0,0 +1,43 @@
+const numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+
+const account = [
+  { type: 'deposit', amount: 150 },
+  { type: 'deposit', amount: 20 },
+  { type: 'withdrawal', amount: 5 },
+  { type: 'deposit', amount: 100 },
+  { type: 'withdrawal', amount: 25 },
+  { type: 'withdrawal', amount: 60 }
+];
+
+const traits = [
+  { color: 'yellow' },
+  { type: 'electric' },
+  { name: 'pikachu' },
+  { level: 15 },
+  { trainer: 'ash' }
+];
+
+const sum = numbers.reduce((previousValue, currentValue) => previousValue + currentValue);
+console.log('sum:', sum);
+
+const product = numbers.reduce((previousValue, currentValue) => previousValue * currentValue);
+console.log('product:', product);
+
+const initialValue = 0;
+
+const balance = account.reduce(function (previousValue, currentValue) {
+  if (currentValue.type === 'withdrawal') {
+    return previousValue - currentValue.amount;
+  } else {
+    return previousValue + currentValue.amount;
+  }
+}, initialValue);
+console.log('balance:', balance);
+
+const composite = traits.reduce(function (previousValue, currentValue) {
+  if (previousValue === 0) {
+    return Object.assign(currentValue);
+  }
```

Comment:
If you didn't pass the second argument of `initialValue` to `reduce` for this one, you wouldn't need this extra check here.